tests/ts_utils/test_rolling_variance.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2025 Cognite AS
+import numpy as np
+import pandas as pd
+import pytest
+
+from indsl.ts_utils.rolling_stats import rolling_variance
+from indsl.exceptions import UserValueError
+
+
+@pytest.mark.core
+@pytest.mark.parametrize(
+    "min_periods, expected",
+    [
+        pytest.param(1, np.array([0.0, 0.5, 1.0, 1.0, 1.0]), id="minimum periods one"),
+        pytest.param(3, np.array([0.0, 0.0, 1.0, 1.0, 1.0]), id="minimum periods three"),
+    ],
+)
+def test_rolling_variance_different_minimum_periods(min_periods, expected):
+    idx = pd.date_range("2022-01-01 00:00:00", periods=5, freq="1min")
+    data = pd.Series([1.0, 2.0, 3.0, 4.0, 5.0], index=idx)
+
+    window = pd.Timedelta(minutes=3)
+    result = rolling_variance(data, time_window=window, min_periods=min_periods)
+
+    np.testing.assert_allclose(
+        result.values,
+        expected,
+        rtol=0,
+        atol=1e-12,
+        err_msg="rolling_variance should match predefined sample variances",
+    )
+
+
+@pytest.mark.core
+def test_rolling_variance_without_time_unit_raises_error():
+    idx = pd.date_range("2022-01-01 00:00:00", periods=5, freq="1min")
+    data = pd.Series(np.arange(5, dtype=float), index=idx)
+
+    with pytest.raises(UserValueError):
```

Comment:
```suggestion
    with pytest.raises(UserValueError, match="Error message here."):
```
it's better to verify what kind of error actually raised